### ISSUE TITLE
are normalize and reset being included in the correct order?

### ISSUE THREAD

**Issue by clindsey** [inuitcss/inuitcss#206]
It seems like some styles defined in `normalize.scss` are being removed in `reset.scss`, such as the `border` of `fieldset`.  **Is this the intended behavior?**

`border` for `fieldset` is added here -
https://github.com/inuitcss/inuitcss/blob/develop/generic/_generic.normalize.scss#L323

`border` for `fieldset` is removed here -
https://github.com/inuitcss/inuitcss/blob/develop/generic/_generic.reset.scss#L50

`normalize.scss` is loaded before `reset.scs`, which removes the `border` definition -
 https://github.com/inuitcss/inuitcss/blob/develop/example.main.scss#L99

**Comment by clindsey** [comment]
oops, missed the comment at the top https://github.com/inuitcss/inuitcss/blob/develop/generic/_generic.reset.scss#L6
```
/**
 * A very simple reset that sits on top of Normalize.css.
 */
```

Can someone clarify the logic behind this `fieldset` `border` thing?

**Comment by herzinger** [comment]
1. This border removal is pretty new, so I think it's a simple mistake.
2. If you gonna use both, I agree reset should be first. But ordinarily you would go with one or the other, right?

**Comment by nenadjelovac** [comment]
@clindsey it is intentional.

See [this PR](https://github.com/inuitcss/inuitcss/pull/193#issue-182491552):

>[...]the browser-default `border` declaration seems a bit too opinionated.

**Comment by herzinger** [comment]
@nenadjelovac no, it's a mistake. It makes no sense to add a border in normalize and remove it in reset. The border rule should be removed from normalize if that's the case.

**Comment by anselmh** [comment]
I think the issue is a different one.

We seem to advertise a reset.css on top of normalize.css. I’m not sure if that’s intended and indeed this makes only little sense, since this involves a lot of resetting of previously set styles. Instead I think, we should either remove reset.css completely or make it separate from normalize as an alternative to it, not as an addition.

Then it’d make sense to have a border set in normalize while not have it set using reset. What do you think?

**Comment by nenadjelovac** [comment]
@herzinger ah, I see the confusion there. You see, there is no scss version of normalize in the official [normalize.css repo](https://github.com/necolas/normalize.css). That's why copy/paste its code into inuitcss. We **do not** make any changes to normalize.

**Comment by csshugs** [comment]
To make one thing clear beforehand: Although the file is named `_generic.reset.scss`, it is not the original [reset.css](http://meyerweb.com/eric/tools/css/reset/) by Eric Meyer. It is a ["very simple reset that sits on top of Normalize.css."](https://github.com/inuitcss/inuitcss/blob/c2474b18235d1af3f5e484fcf711e81811eff2ed/generic/_generic.reset.scss#L6), read: It enhances normalize.css.

@herzinger 

> @nenadjelovac no, it's a mistake. It makes no sense to add a border in normalize and remove it in reset. The border rule should be removed from normalize if that's the case.

It is actually [intentional](https://github.com/inuitcss/inuitcss/pull/193). Due to the lack of an official Sass port for normalize.css, we copy and paste the original normalize.css file into our codebase. Nonetheless, I would consider it 3rd-party code, hence I would not directly edit normalize.css. So I'm against removing the border rule from normalize.css.

What I would rather do is removing [this declaration](https://github.com/inuitcss/inuitcss/blob/c2474b18235d1af3f5e484fcf711e81811eff2ed/generic/_generic.reset.scss#L45-L53) again, because:

a. It's opinionated
b. If someone wants to actually have a border, it would [first get set in normalize.css](https://github.com/inuitcss/inuitcss/blob/c2474b18235d1af3f5e484fcf711e81811eff2ed/generic/_generic.normalize.scss#L324), then [reset in reset.scss](https://github.com/inuitcss/inuitcss/blob/develop/generic/_generic.reset.scss#L52) and then again set in a custom partial by the user.

I guess, the best idea would be to let [this](https://github.com/inuitcss/inuitcss/blob/develop/generic/_generic.reset.scss#L45-L53) do the framework user and remove it.

Without that declaration, to me normalize.css and our reset always made totally sense in conjunction.

**Comment by herzinger** [comment]
>a. It's opinionated
b. If someone wants to actually have a border, it would first get set in normalize.css, then reset in reset.scss and then again set in a custom partial by the user.

@csshugs I agree 100℅, and that's more or less what I said in the issue/pr in question.

About normalize, if that's the case, I think it should be installed as a dependency like SassMq. You can import .css files into sass just fine.

About the reset, I understood it later, and that's why I edited my first response. But the partial name is highly misleading.

**Comment by csshugs** [comment]
@herzinger 

> You can import .css files into sass just fine.

This currently just works with libsass, unfortunately. That's why we cannot install it as a dependency and have to copy/paste it.

> But the partial name is highly misleading.

So we need a new name for it. Any suggestions?

**Comment by herzinger** [comment]
@csshugs I'll try and think of a name.

In the meantime, [this](https://github.com/inuitcss/inuitcss/blob/e67e7541ce5837c5dc77b23c15c64da09d22e2f4/generic/_generic.reset.scss#L52) should be removed and this issue closed, if I follow, right?

**Comment by herzinger** [comment]
To be honest, what our "reset" is is nothing more than a extension of normalize. What we do there wouldn't be out of place as a pull request for the official one. To keep it as it is, as a second partial, we could name it something like "equalize" or "fine-tune", or even embrace that idea and go something like "extend-normalize" or "inuit-normalize" or even name our own "normalize" and import the normal one at the top. There ara a lot of options, so what do you all think? It's need to be decided before I follow with the solution I proposed in #219.

**Comment by csshugs** [comment]
So: I think it's best to leave everything as is right now.

> The reset file exists to reset anything: the fact it doesn’t look like it’s resetting much is not relevant. If we decide we need to reset more things in future we’d just have to create a reset file again.
> 
> — @inuitcss/core member 

---

To answer the initial question of this issue:

We import [normalize.css](https://github.com/necolas/normalize.css/) as 3rd-party module. Unfortunately, [we cannot install it that easy as a dependency](https://github.com/inuitcss/inuitcss/issues/206#issuecomment-256006033) like we do with Sass-MQ. But [we are working on it](https://github.com/inuitcss/inuitcss/issues/219), so we don't need to manually copy/paste this 3rd-party code into our codebase. That being said, since normalize.css is 3rd-party code, we adopt it as-is into our code. That's why we are having the `border` declaration for the `fieldset`—it's coming from 3rd-party code. We have opinionatedly decided to get rid of that `border` in `generic.reset`, because we considered it ... well ... opinionated (The question of removing this override again in `generic.reset` is a topic for another issue; we wont cover this in this issue).

Long story short: **The include order is correct**. We are willingly using normalize.css (in the first place) and enhancing that with further resets in our own reset file (Again, this is not the famous reset.css from Eric Meyer!)
